Marlin/src/module/planner.cpp

```diff
@@ -1230,7 +1294,18 @@ void Planner::_buffer_steps(const int32_t (&target)[ABCE], float fr_mm_s, const
   }
   block->acceleration_steps_per_s2 = accel;
   block->acceleration = accel / steps_per_mm;
-  block->acceleration_rate = (long)(accel * 16777216.0 / (HAL_STEPPER_TIMER_RATE)); // 16777216 = <<24
+  block->acceleration_rate = (long)(accel * 16777216.0 / ((F_CPU) * 0.125)); // * 8.388608
```

Comment:
This change to use `F_CPU` instead of `HAL_STEPPER_TIMER_RATE` breaks things on 32 bit systems. Specifically, the deceleration phase of motion is significantly slower than it should be. Reverting back to the use of `HAL_STEPPER_TIMER_RATE` alleviates the issue.

You might also want to look at the use of `F_CPU` a few lines down as well.

Reply:
Oops, that was a foolish oversight from using paste and not re-checking. I patched it this morning, and the line below.